src/radiation_system.hpp

```diff
@@ -687,6 +693,108 @@ AMREX_GPU_DEVICE auto RadSystem<problem_t>::ComputeCellOpticalDepth(const quokka
 						       // return 0.5*(tau_L + tau_R); // arithmetic mean
 }
 
+template <typename problem_t>
+template <FluxDir DIR, typename TARRAY>
+AMREX_GPU_DEVICE void RadSystem<problem_t>::ComputeRadPressure(const double erad, const double Fx, const double Fy, const double Fz, const double fx,
```

Comment:
This should return the values it computes, rather than modify its arguments. It's much clearer that way. There is no performance reason to do otherwise.